CancerGov/_src/Scripts/NCI/UX/PageSpecific/Inner/Enhancements/directory-search-results.js

```diff
@@ -0,0 +1,105 @@
+const checkedItems = JSON.parse(sessionStorage.getItem('directorySearchList')) || [];
+
+const trackCheckboxes = () => {
+  const checkboxes = document.querySelectorAll(".directory-search-terms ~ div .checkbox input");
+  if (checkboxes.length > 1) {
+    checkboxes.forEach(checkbox => {
+      // if the checkbox value is in session storage then mark it as checked
+      checkbox.checked = checkedItems.indexOf(checkbox.value) > -1?'checked':'';
+      // add click event to checkbox
+      checkbox.addEventListener('click', updateChecklist.bind(this, checkbox));
+    })
+  }
+}
+
+const updateChecklist = (checkbox) => {
+  let item = checkbox.value;
+  let isChecked = checkbox.checked;
+  if (!isChecked) { // Uncheck it
+    if (checkedItems.indexOf(item) > -1) {
+      var index = checkedItems.indexOf(item);
+      checkedItems.splice(index, 1);
+    }
+  } else if ($.inArray(item, checkedItems) == -1) {
+    checkedItems.push(item);
+  }
+  // update the session storage
+  sessionStorage.setItem('directorySearchList', JSON.stringify(checkedItems));
+
+  //console.log(JSON.parse(sessionStorage.getItem('directorySearchList')));
+
+}
+
+
+const bindFormSubmit = () => {
+  // zero out old form submit method added by CDE
+  if(typeof doSubmit == "function") {
+    doSubmit = null;
+  }
+
+  const form = document.getElementById("resultForm");
+  form.addEventListener('submit',handleFormSubmit.bind(this, form));
+
+  // grab the submit button which is not a real submit button, causing form to not be submitable by keyboard
+  const submitishButton = document.querySelector("button.submit");
+  // remove old onclick handler
+  submitishButton.removeAttribute('onclick');
+  // turn submit button into a real submit button which will trigger the form submit event
+  submitishButton.setAttribute('type','submit');
+}
+
+const handleFormSubmit = (form, e) => {
+  e.preventDefault();
```

Comment:
Just as a possibility for an alternate point of approach all the bound eventListener handlers, currying with an arrow function also works great for passing arguments to event handlers without having to use bind. Not a review point, just an alternative to keep in mind. 

eg
const handleFormSubmit = form => e => {
allows:
form.addEventListener('submit', handleFormSubmit.(form));